pkg/operator/ceph/object/admin.go

```diff
@@ -33,17 +33,24 @@ import (
 
 // Context holds the context for the object store.
 type Context struct {
-	Context               *clusterd.Context
-	clusterInfo           *cephclient.ClusterInfo
-	Name                  string
-	UID                   string
-	Endpoint              string
-	Realm                 string
-	ZoneGroup             string
-	Zone                  string
-	adminOpsUserAccessKey string
-	adminOpsUserSecretKey string
-	adminOpsClient        *admin.API
+	Context     *clusterd.Context
+	clusterInfo *cephclient.ClusterInfo
+	Name        string
+	UID         string
+	Endpoint    string
+	Realm       string
+	ZoneGroup   string
+	Zone        string
+}
+
+// AdminOpsContext holds the object store context as well as information for connecting to the admin
+// ops API.
+type AdminOpsContext struct {
+	Context
+	TlsCert               []byte
+	AdminOpsUserAccessKey string
+	AdminOpsUserSecretKey string
+	AdminOpsClient        *admin.API
```

Comment:
if the `AdminOpsClient` have `httpclient` struct have the required tls cert, why do we need to have tlsCert[] here?

Reply:
Initiating the S3 agent in the bucket checker needs the TLS Cert, which isn't stored in a retrievable way in the http.Client AFAICT, so I store the byte info here so it can be retrieved there.